collect_app/src/test/java/org/odk/collect/android/entities/EntitiesRepositoryTest.kt

```diff
@@ -235,12 +277,25 @@ abstract class EntitiesRepositoryTest {
     fun `#delete removes an entity`() {
         val repository = buildSubject()
 
-        val leoville = Entity("wines", "1", "Léoville Barton 2008")
-        val canet = Entity("wines", "2", "Pontet-Canet 2014")
+        val leoville = Entity.New("wines", "1", "Léoville Barton 2008")
+        val canet = Entity.New("wines", "2", "Pontet-Canet 2014")
+        repository.save(leoville, canet)
+
+        repository.delete("1")
+
+        assertThat(repository.getEntities("wines"), containsInAnyOrder(sameEntityAs(canet)))
+    }
+
+    @Test
+    fun `#delete does not change index values for remaining entities`() {
```

Comment:
This actually might not be what we want, but this drives out actually persisting index (as we don't have a query method to test yet). It also felt important to make the interaction between index values and delete explicit.

Would be good to get a second opinion here @lognaturel.

Reply:
Oh actually this totally won't work: multiplicity must start at 0 and not contain any gaps (with the current version of JavaRosa). I'll mark this as a draft again and add that contraint.